tests/test_restrictions_emmo.py

```diff
@@ -0,0 +1,160 @@
+"""Test the API with the EMMO ontology."""
+
+import unittest2 as unittest
+import rdflib
+import itertools
+from osp.core.ontology.entity import OntologyEntity
+from osp.core.ontology.relationship import OntologyRelationship
+from osp.core.ontology.attribute import OntologyAttribute
+from osp.core.ontology.oclass_restriction import Restriction, QUANTIFIER, RTYPE
+from osp.core.ontology.oclass_composition import Composition
+
+
+try:
+    from osp.core.namespaces import math
+except ImportError:  # When the EMMO ontology is not installed.
+    from osp.core.ontology import Parser
+    from osp.core.namespaces import _namespace_registry
+    Parser(_namespace_registry._graph).parse("emmo")
+    _namespace_registry.update_namespaces()
+    math = _namespace_registry.math
+
+
+class TestRestrictionsEmmo(unittest.TestCase):
+    """Test the restrictions in EMMO ontology."""
+
+    emmo_datatypes = (math.Integer, math.Real, math.Boolean, math.Vector,
+                      math.Matrix)
+
+    quantifiers = {rdflib.OWL.someValuesFrom: QUANTIFIER.SOME,
+                   rdflib.OWL.allValuesFrom: QUANTIFIER.ONLY,
+                   rdflib.OWL.cardinality: QUANTIFIER.EXACTLY,
+                   rdflib.OWL.minCardinality: QUANTIFIER.MIN,
+                   rdflib.OWL.maxCardinality: QUANTIFIER.MAX}
+
+    rtypes = {OntologyRelationship: RTYPE.RELATIONSHIP_RESTRICTION,
+              OntologyAttribute: RTYPE.ATTRIBUTE_RESTRICTION}
+
+    def iter_restrictions(self):
+        """Returns all the datatypes of the test wrapped in an iterator."""
+        restrictions = iter([])
+        for datatype in self.emmo_datatypes:
+            restrictions = itertools.chain(restrictions,
+                                           (r for r in datatype.axioms
+                                            if type(r) is Restriction))
+        return restrictions
+
+    def test___str__(self):
+        """Test the __str__ method."""
+        for r in self.iter_restrictions():
+            self.assertIs(type(r.__str__()), str)
```

Comment:
It probably makes sense to also test the specific string value for some specific restrictions